src/Bugsnag.js

```diff
@@ -378,7 +378,8 @@ const allowedMapObjectTypes = ['string', 'number', 'boolean'];
  * Convert an object into a structure with types suitable for serializing
  * across to native code.
  */
-const typedMap = function(map) {
+const typedMap = function(map, maxDepth = 10, depth = 0, seen = new Set()) {
```

Comment:
Are default argument values created globally or at invocation time? In Python, I'd set seen to null and and create inside the body if it is null, as the default argument value is created once and any mutation would propagate.

Reply:
Invocation time: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Evaluated_at_call_time